### ISSUE TITLE
GitHub formatter

### ISSUE THREAD

**Issue by hazzik** [mgechev/revive#1328]
**Is your feature request related to a problem? Please describe.**
When running revive within GitHub Actions workflows, there isn't a built-in formatter that directly produces output compatible with GitHub Actions annotations. This means users need to manually parse the output of existing formatters (like default or json) or use external tools/actions to convert the linting results into annotations that appear directly on pull requests or commit diffs. This adds unnecessary complexity to CI/CD pipelines.

**Describe the solution you'd like**
Introduce a new formatter specifically for GitHub Actions. This formatter, named github, should output each linting failure in the format required by GitHub Actions workflow commands for annotations: `::{{severity}} file={name},line={line},endLine={endLine},title={title}::{message}`. This will allow GitHub Actions to automatically create annotations in the UI based on revive's output.

**Describe alternatives you've considered**
1.  **Manual Parsing/Conversion:** Could pipe the output of an existing formatter (e.g., `json`) to a script (e.g., using `jq` or a custom script) that converts it into the GitHub Actions format. This requires extra scripting and maintenance within the workflow.
2.  **Third-Party Actions:** Using existing marketplace GitHub Actions that wrap `revive` and handle the output formatting (like https://github.com/morphy2k/revive-action). This introduces dependencies on external actions which might have their own limitations or maintenance concerns.

**Additional context**
N/A

**Comment by ccoVeille** [comment]
Thanks for your detailed issue, and the related PR

> Could pipe the output of an existing formatter (e.g., json) to a script (e.g., using jq or a custom script) that converts it into the GitHub Actions format. This requires extra scripting and maintenance within the workflow.

For documentation purpose, do you have any real life example of project doing this?

I tried to find some, but I didn't

Thanks

**Comment by hazzik** [comment]
Sorry, I don't have public examples. We use it in private repository. Here's a snippet from our action:

```yaml
      - name: Run revive
        run: |
          revive -config=revive.toml -formatter=ndjson | \
          jq -r '
            "::\(.Severity) file=\(.Position.Start.Filename),line=\(.Position.Start.Line),endLine=\(.Position.End.Line),title=\(.RuleName)::\(.Failure)"
          '
```

**Comment by hazzik** [comment]
@alexandear thanks for reviewing. What format did you use to get the result? Would it be the default one?

**Comment by hazzik** [comment]
@alexandear thanks for the tip. I've checked and there are 2 formatters that work out of the box with `actions/setup-go` problem matcher: `plain` and `unix`. I'll close this issue.

**Comment by ccoVeille** [comment]
While the issue is now closed, I opened a dedicated issue about this

> We should consider introducing an official GitHub Action for revive itself or marking morphy2k/revive-action as \"an official\".

- https://github.com/mgechev/revive/issues/1332

Because I feel like it should be addressed
